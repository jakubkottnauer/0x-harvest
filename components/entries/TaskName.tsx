import { specialTasks } from "../../lib/api";
import { TaskWithProject } from "../../types";
import { isSoftwareDevTask } from "../../utils";

const TaskName = ({ entry }: { entry: TaskWithProject }) => {
  const specialTask = specialTasks[entry.taskId];

  if (specialTask) {
    return (
      <>
        {specialTask.emoji} {specialTask.displayName || entry.taskName}{" "}
        {specialTask.emoji}
      </>
    );
  }
  if (isSoftwareDevTask(entry.taskName)) {
    // most work projects are called "John - Spotify"
    return <>Work ({entry.projectName.split(" - ")[1] || entry.projectName})</>;
  }
  return <>{entry.taskName}</>;
};

export default TaskName;
