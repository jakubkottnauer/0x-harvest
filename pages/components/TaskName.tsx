import { specialTasks } from "../../lib/api";
import { TimeEntry } from "../../types";
import { isSoftwareDevTask } from "../../utils";

const TaskName = ({ entry }: { entry: TimeEntry }) => {
  const specialTask = specialTasks[entry.task.id];

  if (specialTask) {
    return (
      <>
        {specialTask.emoji} {specialTask.displayName || entry.task.name}{" "}
        {specialTask.emoji}
      </>
    );
  }
  if (isSoftwareDevTask(entry.task.name)) {
    return <>Work ({entry.project.name})</>;
  }
  return <>{entry.task.name}</>;
};

export default TaskName;
