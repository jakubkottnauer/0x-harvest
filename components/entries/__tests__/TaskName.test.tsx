import { render, screen } from "@testing-library/react";
import TaskName from "../TaskName";
import "@testing-library/jest-dom";

describe("TaskName", () => {
  it("renders a correct name for software dev task", () => {
    const { container } = render(
      <TaskName
        entry={{
          projectId: 1,
          projectName: "Test Project",
          taskId: 2,
          taskName: "Software Development",
        }}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("renders a correct name for a regular task", () => {
    const { container } = render(
      <TaskName
        entry={{
          projectId: 1,
          projectName: "Test Project",
          taskId: 2,
          taskName: "Cool Task",
        }}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("renders a correct name for a special task", () => {
    const { container } = render(
      <TaskName
        entry={{
          projectId: 1,
          projectName: "Test Project",
          taskId: 8114249,
          taskName: "Cool Task",
        }}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
