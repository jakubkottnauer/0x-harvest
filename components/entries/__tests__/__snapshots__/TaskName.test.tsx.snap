// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskName renders a correct name for a regular task 1`] = `
<div>
  Cool Task
</div>
`;

exports[`TaskName renders a correct name for a special task 1`] = `
<div>
  ðŸŽ…
   
  Cool Task
   
  ðŸŽ…
</div>
`;

exports[`TaskName renders a correct name for software dev task 1`] = `
<div>
  Work (
  Test Project
  )
</div>
`;
